### ISSUE TITLE
`Form` not sent with same semantics as `FormBody`

### ISSUE THREAD

**Issue by Bilge** [amphp/http-client#333]
I apologise I haven't been able to reduce this down to a simpler test case, but it seems very clear to me that something is not right when sending `Form` in beta 11 as with `FormBody` from the previous API iteration. I haven't been able to narrow down exactly what is different between the two requests, but I can reliably reproduce it with a test.

To test this, please try running [`SteamLoginTest::testSecureLoginCookie`](https://github.com/Provider/Steam/blob/master/test/Functional/SteamLoginTest.php), which requires `STEAM_USER` and `STEAM_PASSWORD` to be set to a valid Steam username and password (you can [get an account](https://store.steampowered.com/join) for free).

`master` will work, but switching over to `connectors-v7`, which has [minimal changes](https://github.com/Provider/Steam/compare/connectors-v7), will fail on `SteamLogin:99` with `{"response":{"interval":5,"extended_error_message":""}}`. Unfortunately this opaque error is all we have to work with.

**Comment by kelunik** [comment]
Some more info would be really helpful. Can you log the full bodies and do a diff there?

**Comment by Bilge** [comment]
I don't know where to capture the full body. Maybe I'll try some hardcore debugging later.

**Comment by Bilge** [comment]
OK I went hardcore, by placing a breakpoint deep within `Http2ConnectionProcessor::request`, and I found that the old (working) code would escape form field content whereas the new (broken) code does not.

For example, base64 characters such as `+` and `/` appear verbatim in the new, broken code whereas they are percent-encoded as `%2B` and `%2F`, respectively, in the old, working code. So that's probably the issue.

Note the headers are the same in both cases (in particular, `content-type` is `application/x-www-form-urlencoded`).

Moreover, note that I cannot manually escape the content because it _will_ encode `%` signs to `%25`, causing them to become double-escaped and thus still fail.
